config/m4/fuse3.m4

```diff
@@ -30,42 +30,40 @@ AS_IF([test "x$with_fuse3" != "xno" -o "x$with_fuse3_static" != "xno"],
               ucx_fuse3_pkgconfig="$with_fuse3/lib$libsuff/pkgconfig"
               export PKG_CONFIG_PATH="$ucx_fuse3_pkgconfig:$PKG_CONFIG_PATH"])
 
-       FUSE3_CPPFLAGS=$(pkg-config --cflags-only-I fuse3)
-       FUSE3_LDFLAGS=$(pkg-config --libs-only-L fuse3)
        AS_IF([test "x$with_fuse3_static" != "xno"],
-             [FUSE3_LIBS=$(pkg-config --libs-only-l --static fuse3)
-              FUSE3_LIBS=${FUSE3_LIBS//"-lfuse3"/"-l:libfuse3.a"}],
-             [FUSE3_LIBS=$(pkg-config --libs-only-l fuse3)])
+             [PKG_CHECK_MODULES_STATIC([FUSE3], [fuse3],
+                                       [fuse3_happy="yes"], [fuse3_happy="no"])])
 
-       save_CPPFLAGS="$CPPFLAGS"
-       save_LDFLAGS="$LDFLAGS"
+       AS_IF([test "x$fuse3_happy" = "xyes"],
+             [FUSE3_LIBS=${FUSE3_LIBS//"-lfuse3"/"-Wl,-Bstatic,-lfuse3,-Bdynamic"}],
+             [PKG_CHECK_MODULES([FUSE3], [fuse3],
+                                [fuse3_happy="yes"], [fuse3_happy="no"])])
+
+       save_CFLAGS="$CFLAGS"
        save_LIBS="$LIBS"
 
-       CPPFLAGS="$FUSE3_CPPFLAGS $CPPFLAGS"
-       LDFLAGS="$FUSE3_LDFLAGS $LDFLAGS"
+       CFLAGS="$FUSE3_CFLAGS $CFLAGS"
        LIBS="$FUSE3_LIBS $LIBS"
 
-       fuse3_happy="yes"
-       AC_CHECK_DECLS([fuse_open_channel, fuse_mount, fuse_unmount],
-                      [AC_SUBST([FUSE3_CPPFLAGS], [$FUSE3_CPPFLAGS])
-                       AC_DEFINE([FUSE_USE_VERSION], 30, [Fuse API version])],
-                      [fuse3_happy="no"],
-                      [[#define FUSE_USE_VERSION 30
-                        #include <fuse.h>]])
+       AS_IF([test "x$fuse3_happy" = "xyes"],
+             [AC_CHECK_DECLS([fuse_open_channel, fuse_mount, fuse_unmount],
+                             [AC_SUBST([FUSE3_CFLAGS], [$FUSE3_CFLAGS])
+                              AC_DEFINE([FUSE_USE_VERSION], 30, [Fuse API version])],
+                             [fuse3_happy="no"],
+                             [[#define FUSE_USE_VERSION 30
+                               #include <fuse.h>]])])
 
        AS_IF([test "x$fuse3_happy" = "xyes"],
              [AC_CHECK_FUNCS([fuse_open_channel fuse_mount fuse_unmount],
-                             [AC_SUBST([FUSE3_LIBS], ["$FUSE3_LIBS"])
-                              AC_SUBST([FUSE3_LDFLAGS], [$FUSE3_LDFLAGS])],
+                             [AC_SUBST([FUSE3_LIBS], ["$FUSE3_LIBS"])],
                              [fuse3_happy="no"])])
 
        AS_IF([test "x$fuse3_happy" != "xyes" -a "x$with_fuse3" != "xguess"],
              [AC_MSG_ERROR([FUSEv3 requested but could not be found])])
 
-       CPPFLAGS="$save_CPPFLAGS"
-       LDFLAGS="$save_LDFLAGS"
+       CFLAGS="$save_CFLAGS"
        LIBS="$save_LIBS"
-       PKG_CONFIG_PATH="$save_PKG_CONFIG_PATH"
+       export PKG_CONFIG_PATH="$save_PKG_CONFIG_PATH"
```

Comment:
@yosefe here is fix for variable manipulation

Reply:
I think we should do only that fix.
I assume Raul tested the existing implementation with different scenarios, why need to risk breaking something and creating ourselves more work?